lib/luahid.c

```diff
@@ -0,0 +1,87 @@
+/*
+* SPDX-FileCopyrightText: (c) 2025 Jieming Zhou <qrsikno@gmail.com>
+* SPDX-License-Identifier: MIT OR GPL-2.0-only
+*/
+
+#define pr_fmt(fmt) KBUILD_MODNAME ": " fmt
+#include <linux/hid.h>
+#include <lunatik.h>
+
+typedef struct luahid_s {
+	lunatik_object_t *runtime;
+	struct hid_driver driver;
+} luahid_t;
+
+static const struct hid_device_id luahid_table[] = {
+	{HID_DEVICE(HID_BUS_ANY, HID_GROUP_ANY, HID_ANY_ID, HID_ANY_ID)},
+	{ }
+};
+MODULE_DEVICE_TABLE(hid, luahid_table);
+
+static void luahid_release(void *private)
+{
+	luahid_t *hid = (luahid_t *)private;
+	if (hid) {
```

Comment:
what exactly are you verifying in here? it's to supposed to always be set for a given hid object.. it's static memory, right? the thing you allocate dynamically is `name`, right? and if it fails to alloc with `lunatik_checkalloc`, you will ending up freeing a "dirty" pointer.. If I got it correctly, you would need to initialize name with NULL, then checkalloc().. and in here you should check if `name != NULL`, right?